hub_test.go

```diff
@@ -117,6 +118,52 @@ func TestNonBlockingSubscriberShouldAlertIfLoseMessages(t *testing.T) {
 	require.Equal(t, []string{"a.*.c"}, msg.Fields["topic"])
 }
 
+func TestCloseBlockingSubscriber(t *testing.T) {
+	h := New()
+
+	sub := h.Subscribe(100, "topic")
+	wg := sync.WaitGroup{}
+
+	go func() {
+		for range sub.Receiver {
+		}
+	}()
+
+	wg.Add(100)
+	for i := 0; i < 100; i++ {
+		go func() {
+			defer wg.Done()
+			h.Publish(Message{Name: "topic"})
+		}()
+	}
+
+	h.Unsubscribe(sub)
```

Comment:
Adding `time.Sleep(time.Nanosecond)` didn't make the test fail every time for me -- and I don't think it should, since the runtime can choose to run the goroutines in any order. One solution would be to run the test with `-count 100` to almost guarantee a failure, what do you think?